src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -161,14 +161,18 @@ class SessionService(
     return validToDate
   }
 
-  private fun filterSessionsTemplatesForLocation(sessionTemplates: List<SessionTemplate>, prisonerId: String?): List<SessionTemplate> {
-    prisonerId?.let { prisonerIdVal ->
-      val prisonerDetailDto = prisonerService.getPrisonerHousingLocation(prisonerIdVal)
-      prisonerDetailDto?.let { prisonerDetail ->
-        val prisonerLevels = prisonerService.getLevelsMapForPrisoner(prisonerDetail)
-        return sessionTemplates.filter { sessionTemplate ->
-          sessionValidator.isSessionAvailableToPrisoner(prisonerLevels, sessionTemplate)
+  private fun filterSessionsTemplatesForLocation(sessionTemplates: List<SessionTemplate>, prisonerId: String?, prisonCode: String): List<SessionTemplate> {
+    val hasSessionsWithLocationGroups = sessionTemplates.any { it.permittedSessionGroups.isNotEmpty() }
+    if (hasSessionsWithLocationGroups) {
+      prisonerId?.let { prisonerIdVal ->
```

Comment:
I realise this is from my earlier code.. so not a code review on your bit but maybe we can do away with the  prisonerIdVal and use prisonerId instead